pithos/pithos.py

```diff
@@ -369,8 +368,10 @@ def init_core(self):
             if is_flatpak():
                 # However in flatpak that path is not readable by the host.
                 tempdir_base = os.path.join(GLib.get_user_cache_dir(), 'tmp')
-            self.tempdir = tempfile.TemporaryDirectory(prefix='pithos-', dir=tempdir_base, ignore_cleanup_errors=True)
-            logging.info("Created temporary directory {}".format(self.tempdir.name))
+            self.tempdir = os.path.join(tempdir_base, 'pithos_cache')
+            if not os.path.exists(self.tempdir):
+                os.makedirs(self.tempdir)
```

Comment:
This is a race condition, just always call `os.makedirs()` with `exist_ok=True`.